bigframes/ml/core.py

```diff
@@ -273,6 +273,8 @@ def _create_model_ref(
     def _create_model_with_sql(self, session: bigframes.Session, sql: str) -> BqmlModel:
         # fit the model, synchronously
         _, job = session._start_query_ml_ddl(sql)
+        if session._metrics is not None:
+            session._metrics.count_job_stats(job)
```

Comment:
Should we not do this inside _start_query_ml_ddl? That way all ML DDLs will be accounted in the session metrics

Reply:
Updated, and added an assertion in test for register.